### ISSUE TITLE
Number parsing

### ISSUE THREAD

**Issue by oleobal** [oleobal/rockstar-interpreter#2]
It seems number are not parsed correctly in some conditions, but I don't understand which exactly as of now.

If you execute `python3 -m pdb rockstar.py scripts/test-blocks.rock`, then `c` to execute, then `u` to go up the call stack after the program crashed, you'll realise the instruction : 
```
If your age is 16
Shout "Sweet Sixteen"
```

Was translated to : 
```
[{'type': 'flow control', 'value': 'If'}, {'type': 'variable', 'value': 'your age'}, {'type': 'comparator', 'value': 'EQ
'}, {'type': 'expression', 'value': [{'type': 'number', 'value': 16.0}]}, {'type': 'number', 'value': 6.0}, {'type': 'bl
ock', 'value': [{'type': 'operator', 'value': 'say'}, {'type': 'string', 'value': 'Sweet Sixteen'}]}]
```

There is a parasitic `{'type': 'number', 'value': 6.0` in there and because it is after the 16, I'm banking on a parsing issue.

Either way, we now try to execute a number, and everything breaks down.

**Comment by oleobal** [comment]
Fixed by 91034dd2efda878b42fc96c01f90a0c937c59dd7

**Comment by Zebralt** [comment]
While fixing issue #6, I realized that both block test files kept failing even with the tokenize fix.
This looks similar to this issue, so I'm posting it here.
It seems there is another problem of phantom trailing numbers when tokenizing numeric literals, this time in a loop it seems (this is when running `test-blocks.rock`) :

      [{'type': 'flow control', 'value': 'If'},
       {'type': 'variable', 'value': 'your age'},
       {'type': 'comparator', 'value': 'EQ'},
       {'type': 'expression', 'value': [{'type': 'number', 'value': 16.0}]},
     > {'type': 'number', 'value': 6.0}, <
       {'type': 'block',
        'value': [[{'type': 'operator', 'value': 'say'}, 
                   {'type': 'string', 'value': 'Sweet Sixteen'}]]}]

I didn't dig up the entire reason why, but `{'type' : 'number', 'value' : 6.0}` is then passed to `processBlock` who just can't understand why it would receive a number as a parameter. 
We should of a default return value for this function rather than not checking input at all.

**Comment by oleobal** [comment]
Fixed by d5516cdc696420708f3cfc01537dd52eca7cee54 .. I hope.
